File: tyh-rag-platform/backend/app/schemas/team.py
"""团队管理 Schema (Pydantic DTO)"""

from typing import Optional
from pydantic import BaseModel


# ==================== 团队 CRUD ====================

class TeamCreate(BaseModel):
    """创建团队"""
    name: str
    description: Optional[str] = None


class TeamUpdate(BaseModel):
    """更新团队"""
    name: Optional[str] = None
    description: Optional[str] = None


class TeamResponse(BaseModel):
    """团队响应"""
    id: str
    name: str
    description: Optional[str] = None
    member_count: int = 0
    has_assistant: bool = False
    dataset_count: int = 0
    created_at: str
    updated_at: str


class TeamListResponse(BaseModel):
    """团队列表响应"""
    items: list[TeamResponse]
    total: int
    page: int
    page_size: int


# ==================== 成员管理 ====================

class MemberAdd(BaseModel):
    """添加成员"""
    user_ids: list[str]
    set_default: bool = False


class MemberResponse(BaseModel):
    """成员响应"""
    user_id: str
    username: str
    display_name: str
    role: str
    is_default: bool
    joined_at: str


class MemberListResponse(BaseModel):
    """成员列表响应"""
    items: list[MemberResponse]
    total: int


# ==================== 团队配置（助手绑定） ====================

class ConfigUpdate(BaseModel):
    """绑定助手"""
    ragflow_assistant_id: str


class ConfigResponse(BaseModel):
    """团队配置响应"""
    team_id: str
    ragflow_assistant_id: Optional[str] = None
    ragflow_assistant_name: Optional[str] = None


# ==================== 知识库绑定 ====================

class DatasetBind(BaseModel):
    """绑定知识库（全量替换）"""
    dataset_ids: list[str]


class DatasetItem(BaseModel):
    """知识库条目"""
    id: str
    ragflow_dataset_id: str
    ragflow_dataset_name: Optional[str] = None
    document_count: int = 0
    chunk_count: int = 0


class DatasetListResponse(BaseModel):
    """团队知识库列表响应"""
    items: list[DatasetItem]
    total: int


# ==================== 团队切换 ====================

class ActiveTeamSwitch(BaseModel):
    """切换活跃团队"""
    team_id: str


class MyTeamItem(BaseModel):
    """用户所属团队条目"""
    team_id: str
    team_name: str
    is_default: bool
    is_active: bool


class MyTeamsResponse(BaseModel):
    """用户所属团队列表响应"""
    items: list[MyTeamItem]
    active_team_id: Optional[str] = None

File: tyh-rag-platform/frontend/src/types/team.ts
/**
 * 团队管理相关类型定义
 */

/** 团队基本信息 */
export interface Team {
    id: string;
    name: string;
    description?: string;
    member_count: number;
    has_assistant: boolean;
    dataset_count: number;
    created_at: string;
    updated_at: string;
}

/** 团队成员 */
export interface TeamMember {
    user_id: string;
    username: string;
    display_name: string;
    role: string;
    is_default: boolean;
    joined_at: string;
}

/** 团队配置（助手绑定） */
export interface TeamConfig {
    team_id: string;
    ragflow_assistant_id?: string;
    ragflow_assistant_name?: string;
}

/** 团队知识库绑定 */
export interface TeamDataset {
    id: string;
    ragflow_dataset_id: string;
    ragflow_dataset_name?: string;
    document_count: number;
    chunk_count: number;
}

/** RAGFlow 助手（代理接口返回） */
export interface RagflowAssistant {
    id: string;
    name: string;
}

/** RAGFlow 知识库（代理接口返回） */
export interface RagflowDataset {
    id: string;
    name: string;
    description?: string;
    document_count: number;
    chunk_count: number;
}

/** 用户所属团队条目 */
export interface MyTeamItem {
    team_id: string;
    team_name: string;
    is_default: boolean;
    is_active: boolean;
}

/** 用户团队列表响应 */
export interface MyTeamsResponse {
    items: MyTeamItem[];
    active_team_id?: string;
}
